articles/azure-functions/durable-functions-disaster-recovery-geo-distribution.md

```diff
@@ -0,0 +1,81 @@
+
+
+>[!NOTE]
+>All function apps that are redundantly deployed, must share the same function access keys in the case of being activated via HTTP. The Functions Runtime exposes a [management API](https://github.com/Azure/azure-functions-host/wiki/Key-management-API) that enables consumers to programmatically add, delete, and update function keys.
+>
+
+## Scenario 1 - Load Balanced Compute with Shared Storage
+In the event of problems of the compute service in Azure, the Function App could be affected. To minimize the possibility of such downtime, this scenario uses two instances of the Function App deployed to different regions. The underlying storage account and task hub are created in the main region, and it is shared by both instances.
+Traffic Manager is configured to detect problems in the main site and automatically redirect traffic to the healthy Function App in the failover region.
+Because Function App the failover region shares the same Azure Storage account and Task Hub, the state of the functions is not lost, and word can resume normally. Once the health is restored in the main region, Traffic Manager will start routing requests to that instance of the Function App automatically.
+
+
+![Diagram showing scenario 1.](media/durable-functions-disaster-recovery-geo-distribution/durable-functions-geo-scenario01.png)
+
+There are several benefits when using this deployment scenario:
+- In the case of a failure in the compute layer, work can resume in the failover region without state loss.
+- Traffic Manager takes care of the automatic failover to the healthy instance automatically.
+- Traffic Manager automatically re-establishes traffic to the main Function App after the outage has been corrected.
+
+However, consider the following when using this scenario:
+- If the Function App is deployed using a dedicated AppService plan, there is an increased cost by replicating the compute instance in the failover datacenter.
+- This scenario covers outages at the compute layer, but the storage account continues to be the single point of failure for the Function App. If there is a Storage outage, the application suffers a downtime.
+- If the FunctionApp is failed over, there will be increased latency since it will access its storage account across regions.
+- Accessing the storage service from a different region where it is located incurs in higher cost due to network egress traffic.
+- This scenario depends on Traffic Manager. Considering [how Traffic Manager works](../traffic-manager/traffic-manager-overview#how-traffic-manager-works), it may be some time until a client application that consumes a Durable Function needs to re-query the FunctionApp address from Traffic Manager. 
+
+
+## Scenario 2 - Load Balanced Compute with Regional Storage
+The previous proposed scenario covers only the case of failure in the compute services. If the storage service fails, it will result in an outage of the FunctionApp.
```

Comment:
The previous proposed scenario --> The previous**ly** proposed scenario

Reply:
Or "The preceding scenario..."

Reply:
check